xcode/Subconscious/Shared/Components/Common/Forms/ValidatedTextField.swift

```diff
@@ -7,17 +7,56 @@
 
 import SwiftUI
 import Combine
+import ObservableStore
+
+struct ValidatedFormField<T: Equatable, Action>: View {
+    var alignment: HorizontalAlignment = .leading
+    var placeholder: String
+    var field: FormField<String, T>
+    var send: (Action) -> Void
+    var tag: (FormFieldAction<String>) -> Action
```

Comment:
I would like to avoid passing down tagging functions, and instead pass down a tagged `send` function. The tagging function is an implementation detail the child shouldn't know about, and we don't want the child to be able to pass up anything except its own actions.

Eventually this tagging pattern will be replaced by `StoreProtocol.viewStore` helper https://github.com/subconsciousnetwork/ObservableStore/blob/main/Sources/ObservableStore/ObservableStore.swift#LL347C17-L347C26

Reply:
I'm not sure why I blanked on the fact we already have `Address.forward`. I switched everything to use that instead.